torch/testing/_internal/common_methods_invocations.py

```diff
@@ -4365,6 +4372,16 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
                # cuda gradchecks are slow
                # see discussion https://github.com/pytorch/pytorch/pull/47761#issuecomment-747316775
                SkipInfo('TestGradients', 'test_fn_gradgrad', device_type='cuda'),)),
+    OpInfo('isin',
+           dtypesIfCPU=all_types(),
+           dtypesIfCUDA=all_types_and(torch.half),
+           supports_autograd=False,
+           sample_inputs_func=sample_inputs_isin,
+           skips=(
+               # Skip unsupported checks for bool and complex
```

Comment:
and bfloat16 ... but why skip these checks?

Reply:
We should chat more about this on a VC or something, but the TL;DR is that the generic composite approach implemented for CUDA actually supports these types, causing the respective unsupported tests to fail.

More generally, I'm interested in the right way to handle partial support for some types that comes about accidentally.

Reply:
As discussed in the VC, I added checks to guard against unsupported input dtypes consistently.